pyproject.toml

```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["hatchling", "jupyterlab<4", "hatch-nodejs-version"]
+requires = ["hatchling", "jupyterlab==4.0.9", "hatch-nodejs-version"]
```

Comment:
Can you relax the pin to accept greater or equal (here and in other places)? For one, there were security fixes since the v4.0.9 came out so we should not install the insecure version by default (not that it has much impact on the ephemeral environment used with isolated build, but still it could perpetuate unnecessary pin to the outdated version).

Reply:
I have relaxed it now. I am getting the following error in the CI (and locally) after running(`jupyter labextension list`):

```
JupyterLab v4.1.2
/opt/hostedtoolcache/Python/3.11.8/x64/share/jupyter/labextensions
        jupyterlab_pygments v0.3.0 enabled OK (python, jupyterlab_pygments)
        jupyterlab-conda-store v2024.1.1 enabled  X (python, jupyterlab-conda-store)


   The following extensions are outdated:
        jupyterlab-conda-store
        
   Consider checking if an update is available for these packages.

+ jupyter labextension list
+ grep -ie 'jupyterlab-conda-store.*OK'
```

Any idea, how can I debug this? It's been so long that I put up this PR, that I don't remember a lot of things.

Reply:
It has a point but the error message is poor (https://github.com/jupyterlab/jupyterlab/issues/14351). The problem is the mismatch of React dependencies against JupyterLab - JupyterLab 4.0 and newer pins to v18.2.0 or newer, but the `package.json` contains the old ^17.0.1 version. Not sure why it was passing before but it should not be.

```
[('react', '^18.2.0', '^17.0.1'), ('react-dom', '^18.2.0', '^17.0.1')]
```

Reply:
Oh, that explains why! Thanks, just updated and pushed.